dotnet/src/Connectors/Connectors.OpenAI/OpenAIPromptExecutionSettings.cs

```diff
@@ -287,6 +287,23 @@ public int ResultsPerPrompt
         }
     }
 
+    /// <summary>
+    /// An abstraction of additional settings for chat completion, see https://learn.microsoft.com/en-us/dotnet/api/azure.ai.openai.azurechatextensionsoptions.
+    /// This property is compatible only with Azure OpenAI.
+    /// </summary>
+    [Experimental("SKEXP0010")]
+    [JsonIgnore]
+    public AzureChatExtensionsOptions? AzureChatExtensionsOptions
```

Comment:
I wonder whether an abstraction layer over the `AzureChatExtensionsOptions` class was considered as an option to isolate service API consumers from internal implementation details?

Reply:
Great question! I was thinking about it, but the number of possible configuration properties for this functionality is huge, here is just some of them:
![image](https://github.com/microsoft/semantic-kernel/assets/13853051/ed4a63c6-b443-4ebb-a9c4-df835b3aba29)

I think it would be difficult to abstract these parameters and maintain them if anything changes in the future. I also think that abstraction could be useful if similar settings would exist in other connectors as well, but as for now it's Azure-specific settings, so I decided to expose `AzureChatExtensionsOptions` class directly, in similar way how we do that with `OpenAIClient` in Azure/OpenAI chat completion service constructors.

Let me know what you think about it.

Reply:
I think this solution is good for now, considering the large number of classes that require abstraction to achieve the isolation we might never need. As you mentioned, let's revise the model later when we have more connectors using similar settings.